### PR TITLE
code-intel: allow excluding vertices from lsif-visualize output

### PR DESCRIPTION
**Pull request by Strum355**
Does what it says on the tin. Also excludes the edges with outV or inVs pointing to excluded vertex

### CODE REVIEW

File: enterprise/lib/codeintel/lsif/test/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+lsif-visualize
```
Comment (efritz) [inline enterprise/lib/codeintel/lsif/test/.gitignore:1]: This ignores the lsif-visualize directory as well so new files won't get added directly.

File: enterprise/lib/codeintel/lsif/test/cmd/lsif-visualize/args.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ var (
 	indexFile     *os.File
 	fromID        int
 	subgraphDepth int
+	exclude       = make([]string, 0)
```
Comment (efritz) [inline enterprise/lib/codeintel/lsif/test/cmd/lsif-visualize/args.go:18]: No point in making a slice here, nil is fine if it has no capacity

**Review by efritz — APPROVED** [review]
qq: Looks like this excludes exact vertices, but may still show transitively reachable vertices that are now a disconnected graph?

**Comment by Strum355** [comment]
> but may still show transitively reachable vertices that are now a disconnected graph?
yes thats the case currently, its a start at least (exclude monikers without manually patching the file)

**Pull request merged**